cmd/cluster-displayer/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/util"
+)
+
+type Page struct {
+	Data []map[string]string `json:"data"`
+}
+
+func gatherOptions() (options, error) {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	fs.IntVar(&o.port, "port", 8081, "Port to run the server on")
+	fs.StringVar(&o.hiveKubeconfigPath, "hive-kubeconfig", "", "Path to the kubeconfig file to use for requests to Hive.")
+	fs.DurationVar(&o.gracePeriod, "gracePeriod", time.Second*10, "Grace period for server shutdown")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return o, fmt.Errorf("failed to parse flags: %w", err)
+	}
+	return o, nil
+}
+
+func validateOptions(o options) error {
+	_, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %w", err)
+	}
+	if o.hiveKubeconfigPath == "" {
+		return fmt.Errorf("--hive-kubeconfig must be set")
+	}
+	return nil
+}
+
+type options struct {
+	logLevel           string
+	port               int
+	gracePeriod        time.Duration
+	hiveKubeconfigPath string
+}
+
+func addSchemes() error {
+	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
+		return fmt.Errorf("failed to add hivev1 to scheme: %w", err)
+	}
+	return nil
+}
+
+func getLabelsString(l map[string]string) string {
+	keys := make([]string, 0, len(l))
+	for k := range l {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	var s []string
+	for _, k := range keys {
+		s = append(s, fmt.Sprintf("%s=%s", k, l[k]))
+	}
+	strings.Join(s, ",")
+	return strings.Join(s, ",")
+}
+
+func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client) (*Page, error) {
+	clusterPools := &hivev1.ClusterPoolList{}
+	if err := hiveClient.List(ctx, clusterPools); err != nil {
+		return nil, fmt.Errorf("failed to list cluster pools: %w", err)
+	}
+	var page Page
```

Comment:
If you don't want the zero value (e.g. for the data) just use a literal:

```go
page := Page{
    Data: []map[string]string{}
}
```

Reply:
Updated.